src/test/java/org/junit/tests/assertion/AssertionTest.java

```diff
@@ -34,6 +34,10 @@
 //      assert false;
 //  }
 
+    private void failAssertionErrorExpected() {
+        throw new AssertionError("AssertionError expected");
```

Comment:
It's intentional that exception is thrown directly here, as this class also have tests for the `fail()` method. So it not using fail() here in this particular case is a good thing :-)